src/index.js

```diff
@@ -1,14 +1,125 @@
+const ul = document.getElementById('todo-list');
+let li;
 
-  // Lodash, now imported by this script
-  element.innerHTML = _.join(['Hello', 'ahmed'], ' ');
-  element.classList.add('hello');
+let tasks = [];
 
-  return element;
+function getTasks() {
+  if (localStorage.getItem('tasks') == null) {
+    tasks = [];
+  } else {
+    tasks = Array.from(JSON.parse(localStorage.getItem('tasks')));
+  }
+  return tasks;
 }
 
-document.body.appendChild(component());
+function prepareEdit(task, btn) {
+  btn.addEventListener('click', () => {
+    const el = document.querySelector(`[data-index="${task.index}"]`);
+    const p = el.children[1];
+    const input = document.createElement('input');
+    input.type = 'text';
+    input.value = p.textContent;
+    input.id = task.index;
+    el.insertBefore(input, p);
+    el.removeChild(p);
+    input.focus();
+    input.select();
+    input.classList.add('edit');
+  });
+}
+
+function loadTasks() {
+  const tasks = Array.from(JSON.parse(localStorage.getItem('tasks')));
+  tasks.forEach((task) => {
+    li = document.createElement('li');
+    li.classList.add('task-li');
+    li.innerHTML = `
+        <div class="task" data-index="${task.index}">
+          <input type="checkbox">
+          <p>${task.description}</p>
+          <div class="btns">
+            <button type="button" class="close-button scroll">+</button>
+            <button type="button" class="fas fa-ellipsis-v scroll" id="edit-${task.index}"></button>
+          </div>
+        </div>
+    `;
+    ul.appendChild(li);
+    const btn = document.getElementById(`edit-${task.index}`);
+    prepareEdit(task, btn);
+    localStorage.setItem('tasks', JSON.stringify(tasks));
+    // getTasks();
+  });
+}
+
+function addTask() {
+  ul.innerHTML = '';
+  tasks.forEach((task) => {
+    li = document.createElement('li');
+    li.classList.add('task-li');
+    li.innerHTML = `
+        <div class="task" data-index="${task.index}">
+          <input type="checkbox">
+          <p>${task.description}</p>
+          <div class="btns">
+            <button type="button" class="close-button scroll">+</button>
+            <button type="button" class="fas fa-ellipsis-v scroll" id="edit-${task.index}"></button>
+          </div>
+        </div>
+    `;
+    ul.appendChild(li);
+    const btn = document.getElementById(`edit-${task.index}`);
+    prepareEdit(task, btn);
+    localStorage.setItem('tasks', JSON.stringify(tasks));
+    // getTasks();
+    console.log(tasks);
+  });
+}
+
+function removeTask(e) {
+  if (e.target.classList.contains('close-button')) {
+    const deletedIndex = e.target.parentElement.parentElement.dataset.index;
+    const toBeUpdatedTasks = tasks.slice(deletedIndex, tasks.length);
+    toBeUpdatedTasks.forEach((task) => {
+      const el = document.querySelector(`[data-index="${task.index}"]`);
+      el.dataset.index = task.index - 1;
+      task.updatedIndex = task.index - 1;
+    });
```

Comment:
- [ ] I can see that you are trying to implement changes from the previous reviewer. You do not need to add an `updatedIndex` key when you delete an item, all that is required for you is to rearrange the values of the `index` key of each object in the array.

_**Hint**_:
```
const filteredArray = array.filter((obj) => obj.completed !== true);
    filteredArray.forEach((obj, index) => {
      obj.index = index + 1;
    })
```

Reply:
👍🏽 This works fine now